source/effect_symbol_table_intrinsics.inl

```diff
@@ -1801,10 +1826,27 @@ IMPLEMENT_INTRINSIC_SPIRV(tex2DgatherR, 1, {
 		.add(args[2].base)
 		.result;
 	})
+IMPLEMENT_INTRINSIC_SPIRV(tex2DgatherR, 2, {
+	add_capability(spv::CapabilityImageGatherExtended);
+
+	const spv::Id component = emit_constant(0u);
+	return add_instruction(spv::OpImageGather, convert_type(res_type))
+		.add(args[0].base)
+		.add(args[1].base)
+		.add(component)
+		.add(spv::ImageOperandsConstOffsetsMask)
+		.add(args[2].base)
```

Comment:
Looks like `ImageOperandsConstOffsetsMask` accepts a single argument, which should be an array, rather than multiple separate arguments (`Offsets must be an <id> of a constant instruction making an array of size four of vectors of two integer components.`):
Quick test with glslang indeed generates this:
```
%35 = OpConstantComposite %_arr_v2int_uint_4 %25 %28 %31 %34
%36 = OpImageGather %v4float %14 %17 %int_3 ConstOffsets %35
```
So I believe this should to the job:
```c++
const spv::Id component = emit_constant(0u);
const spv::Id offsets = add_instruction(spv::OpConstantComposite, convert_type({ reshadefx::type::t_int, 2, 1, 0, 4 }), _types_and_constants)
	.add(args[2].base)
	.add(args[3].base)
	.add(args[4].base)
	.add(args[5].base)
	.result;
return add_instruction(spv::OpImageGather, convert_type(res_type))
	.add(args[0].base)
	.add(args[1].base)
	.add(component)
	.add(spv::ImageOperandsConstOffsetsMask)
	.add(offsets)
	.result;
```